integrations/defectdojo-JSONCreatedUsingSemgrepAPI/transformFindingsToJSON.py

```diff
@@ -0,0 +1,93 @@
+import requests
+import json
+import os
+from uploadSemgrepJSONToDefectDojo import uploadToDefectDojo
+import sys
+# Configuration
+deployment_slug = "swati"
+BASE_URL = "https://semgrep.dev"
+findings_url = BASE_URL+"/api/v1/deployments/"+deployment_slug+"/findings"
+SEMGREP_APP_TOKEN = os.environ["SEMGREP_API_TOKEN"]
+OUTPUT_FILE = "semgrep_findings_formatted.json"
+dd_url= "http://localhost:8080"
+product_name = "Semgrep SAST"
+engagement_name = "Semgrep Scans"
+dd_token = os.getenv("DEFECT_DOJO_API_TOKEN")
+def retrieve_paginated_data(endpoint):
+    all_findings = []
+    page = 0
+    page_size = 100  # Adjust as needed
+    headers = {"Accept": "application/json", "Authorization": "Bearer " + SEMGREP_APP_TOKEN}
+    while True:
+        response = requests.get(endpoint+"?&page_size="+str(page_size)+"&page="+str(page), headers=headers)
+        
+        if response.status_code != 200:
+            print(f"Error: {response.status_code}, {response.text}")
+            break  # Stop on failure
+        try:
+            data = response.json()
+        except json.JSONDecodeError:
+            print(f"Failed to parse JSON: {response.text}")
+            break
+
+        findings = data["findings"]
+         # If findings list is empty, stop fetching more pages
+        if not findings:
+            print("No findings available.")
```

Comment:
Since this message will come up even if some findings have already been fetched, I would say "no more findings" or something of that nature.